### PR TITLE
Support tarball installation, small improve of PR #44

### PR DESCRIPTION
**Pull request by yyvess**
Improve PR https://github.com/puppetlabs/puppetlabs-activemq/pull/44
- Fix conflicts
- Keep defaults for _mq_admin_username_ and _mq_admin_username_
- Fix https support on binary url
- Modify log directory to /var/log/activemq
- Do not warn about default password when a external configuration template is used
- Adding sample on README

### CODE REVIEW

**Comment by gehel** [comment]
You should probably send this PR directly upstream.

File: manifests/config.pp [hunk 1]
```diff
@@ -11,33 +11,35 @@
...
 #
 class activemq::config (
-  $server_config,
-  $instance,
-  $package,
-  $path = '/etc/activemq/activemq.xml',
-  $server_config_show_diff = 'UNSET',
+  $server_config           = $activemq::server_config_content,
+  $instance                = $activemq::instance,
+  $package                 = $activemq::package,
```
Comment (zonArt) [inline manifests/config.pp]: This one (`$package`) is useless as it is never used, either keep it and let the `require => Package[$package]` or remove it and then you can change the require as below. Personally I tend to prefer the first solution

File: manifests/service.pp [hunk 2]
```diff
@@ -11,21 +11,23 @@
...
 
   validate_bool($service_enable)
 
-  $ensure_real = $ensure
-
   service { 'activemq':
-    enable     => $service_enable,
-    ensure     => $ensure_real,
+    ensure     => $ensure,
+    enable     => $enable,
```
Comment (zonArt) [inline manifests/service.pp:28]: This introduce a change in behavior and links the ensure argument with the fact that the service is active or not (which wasn't the case before). Service can be present but disabled.

Moreover with this modification the `$service_enable` param doesn't make sense anymore

File: manifests/config.pp (hunk 1 above)
Comment (yyvess) [inline manifests/config.pp]: I will remove it. 
`require => Package[$package]` can not be used when we install activemq from a `tar`.

File: manifests/service.pp (hunk 2 above)
Comment (yyvess) [inline manifests/service.pp:28]: You are true, I remove it.
Thanks

**Pull request closed without merging**